### PR TITLE
tests: Make syncval to pass core validation

### PR DESCRIPTION
**Pull request by artem-lunarg**
* Fix syncval tests to work with core validation
* `--syncval-enable-core` command line option to conveniently test syncval+core during development
* Add pass to CI  (thanks Spencer for the guide) to run syncval with core validation enabled (in addition to regular syncval only)

### CODE REVIEW

File: layers/core_checks/cc_cmd_buffer.cpp [hunk 1]
```diff
@@ -326,8 +326,6 @@ bool CoreChecks::PreCallValidateEndCommandBuffer(VkCommandBuffer commandBuffer,
     const CMD_BUFFER_STATE &cb_state = *cb_state_ptr;
     if ((VK_COMMAND_BUFFER_LEVEL_PRIMARY == cb_state.createInfo.level) ||
         !(cb_state.beginInfo.flags & VK_COMMAND_BUFFER_USAGE_RENDER_PASS_CONTINUE_BIT)) {
-        // This needs spec clarification to update valid usage, see comments in PR:
```
Comment (artem-lunarg) [inline layers/core_checks/cc_cmd_buffer.cpp:329]: Addressed by spec:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/165#issuecomment-449047637

File: tests/unit/sync_val.cpp [hunk 2]
```diff
@@ -3182,6 +3182,10 @@ TEST_F(NegativeSyncVal, RenderPassAsyncHazard) {
             m_errorMonitor->VerifyFound();
         }
 
+        // Suppress core validation that render pass ends before reaching final subpass.
+        // NextSubpass does not update current subpass (Record is skipped) due to syncval error.
+        m_errorMonitor->SetUnexpectedError("UID-vkCmdEndRenderPass-None-00910");
```
Comment (spencer-lunarg) [inline tests/unit/sync_val.cpp:3187]:
```suggestion
        m_errorMonitor->SetUnexpectedError("VUID-vkCmdEndRenderPass-None-00910");
```

**Review by spencer-lunarg — APPROVED** [review]
looks awesome, simple way to both test in CI and locally, love it :rocket:

File: tests/unit/sync_val.cpp (hunk 2 above)
Comment (artem-lunarg) [reply tests/unit/sync_val.cpp:3187]: Thanks for catching this. It looks `SetUnexpectedError` checks for substrings.

Comment (spencer-lunarg) [reply tests/unit/sync_val.cpp:3187]: it does, but just to save the urge of someone later wanting to fix it

Comment (artem-lunarg) [reply tests/unit/sync_val.cpp:3187]: Fixed. That's copy pasted part, fixed one, forgot about this. Waiting for AI to help with such issues.

**Pull request merged**